### PR TITLE
Update DG - Implementation and Use Cases

### LINKED ISSUE #121: Update DG - Implementation and Use Cases

### PR DESCRIPTION
**Pull request by rolandyuwy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -216,33 +216,47 @@ The following activity diagram summarizes what happens when a user executes a ne
...
-When the commands are executed by calling `SortExpiryCommand#execute(Model model)` or `SortPriorityCommand#execute(Model model)`, the `SortedList<Food>` attribute in `model` is sorted.
+Similarly, `SortPriorityCommand` sorts the list of food displayed by priority from `HIGH` to `LOW`, followed by expiry date from oldest to newest, followed by description.
 
-This is done so by calling `model#updateSortedFoodList(Comparator<Food> comparator)` method in `model` with the relevant `Comparator<Food>` for sorting.
+When the commands are executed by calling `SortDescCommand#execute(Model model)` or `SortExpiryCommand#execute(Model model)` or `SortPriorityCommand#execute(Model model)`, the `versionedFoodInventory` attribute in `model` is sorted.
 
-Sorting of the `SortedList<Food>` attribute in `model` is reflected in the GUI when `MainWindow` calls `logic#getFilteredFoodList()`.
+This is done so by calling `model#sortFoodInventory(Comparator<Food>... comparators)` method in `model` with the relevant `comparators` for sorting.
+
+`model#setSortingComparators` and `userPref#setSortingComparatorsDescription` are then called to save the sorting information.
```
Comment (bchenghi) [inline docs/DeveloperGuide.md]: Should we have brackets after the method names, for consistency? e.g. `model#setSortingComparators(Comparator<Food>[] sortingComparators)`

**Review by bchenghi — COMMENTED** [review]
LGTM! I just added one comment.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (rolandyuwy) [reply docs/DeveloperGuide.md]: Will do, thanks!

**Review by bchenghi — APPROVED** [review]
LGTM

**Pull request merged**